internal/aws/ecssd/filter.go

```diff
@@ -0,0 +1,90 @@
+// Copyright  OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ecssd
+
+import (
+	"fmt"
+	"sort"
+
+	"go.uber.org/multierr"
+	"go.uber.org/zap"
+)
+
+type TaskFilter struct {
+	logger   *zap.Logger
+	matchers map[MatcherType][]Matcher
+}
+
+type TaskFilterOptions struct {
+	Logger   *zap.Logger
+	Matchers map[MatcherType][]Matcher
+}
```

Comment:
want to learn why we can't just use one structure if it is exactly the same -_-

Reply:
We can, I just removed the options struct and pass the parameter directly in the new func (\w/)